parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -175,12 +188,53 @@ pub mod pallet {
 	pub(super) type ExecutionHeaders<T: Config> =
 		StorageMap<_, Identity, H256, ExecutionHeader, OptionQuery>;
 
+	/// Execution headers ring buffer map implementation
+
+	/// Index storage for execution header ring buffer map
+	#[pallet::storage]
+	pub(crate) type ExecutionHeaderBufferIndex<T: Config> = StorageValue<_, u64, ValueQuery>;
+
+	/// Intermediate storage for execution header mapping
+	#[pallet::storage]
+	pub(crate) type ExecutionHeaderMapping<T: Config> =
+		StorageMap<_, Identity, u64, H256, ValueQuery>;
+
+	/// Ring buffer Map for Execution header
+	pub(crate) type ExecutionHeaderRingBufferMap<T> = RingBufferMapImpl<
```

Comment:
I generally prefer less verbose naming:

```suggestion
	pub(crate) type ExecutionHeadersIndex<T: Config> = StorageValue<_, u64, ValueQuery>;

	/// Intermediate storage for execution header mapping
	#[pallet::storage]
	pub(crate) type ExecutionHeadersMapping<T: Config> =
		StorageMap<_, Identity, u64, H256, ValueQuery>;

	/// Ring buffer Map for Execution header
	pub(crate) type ExecutionHeadersBuffer<T> = RingBufferMapImpl<
```

Reply:
This is the name that will be `STORAGE_PREFIX` and would be displayed in UI. So, its better to be verbose with the naming to give user precise idea about what exactly particular storage entry is used for.

Reply:
I prefer the concise.  Though it's still possible to add some storage macro ```#[pallet::storage_prefix = "***RingBuffer"]``` to explicit the prefix if nessessary IMO it's some internal detail end user may not really care.

Reply:
yeah I still prefer the concise form. Its only developers who would be using the Polkadot-JS console anyway, and they can refer to the comments in the code/docs to see what the storage fields are about.

Reply:
Make sense. Let's make it concise.